packages/editor/src/components/autosave-monitor/test/index.js

```diff
@@ -29,128 +25,167 @@ describe( 'AutosaveMonitor', () => {
+
 	it( 'should start autosave timer after being mounted', () => {
+		render( <AutosaveMonitor isDirty /> );
+
 		expect( setAutosaveTimerSpy ).toHaveBeenCalled();
 	} );
 
 	it( 'should clear the autosave timer after being unmounted', () => {
-		wrapper.unmount();
+		const { rerender } = render( <AutosaveMonitor isDirty /> );
+
+		rerender( <div /> );
+
+		expect( clearTimeout ).toHaveBeenCalled();
+	} );
+
+	it( 'should clear and restart autosave timer when the interval changes', () => {
+		const { rerender } = render( <AutosaveMonitor isDirty /> );
+
+		rerender( <AutosaveMonitor isDirty interval={ 999 } /> );
+
 		expect( clearTimeout ).toHaveBeenCalled();
+		expect( setAutosaveTimerSpy ).toHaveBeenCalledTimes( 2 );
 	} );
 
-	describe( '#componentDidUpdate()', () => {
-		it( 'should clear and restart autosave timer when the interval changes', () => {
-			wrapper.setProps( { interval: 999 } );
-			expect( clearTimeout ).toHaveBeenCalled();
-			expect( setAutosaveTimerSpy ).toHaveBeenCalledTimes( 2 );
-		} );
-
-		it( 'should set needsAutosave=true when editReference changes', () => {
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-			wrapper.setProps( {
-				editsReference: [],
-			} );
-			expect( wrapper.instance().needsAutosave ).toBe( true );
-		} );
-
-		it( 'should set needsAutosave=true when editReference changes and the post becomes dirty', () => {
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-			wrapper.setProps( {
-				isDirty: true,
-				editsReference: [],
-			} );
-			expect( wrapper.instance().needsAutosave ).toBe( true );
-		} );
-
-		it( 'should not set needsAutosave=true when editReference changes and the post is not dirty anymore', () => {
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-			wrapper.setProps( {
-				isDirty: true,
-				editsReference: [],
-			} );
-			wrapper.setProps( {
-				isDirty: false,
-				editsReference: [],
-			} );
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-		} );
-
-		it( 'should set needsAutosave=true when editReference changes and the post is not autosaving', () => {
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-			wrapper.setProps( {
-				isAutosaving: false,
-				editsReference: [],
-			} );
-			expect( wrapper.instance().needsAutosave ).toBe( true );
-		} );
-
-		it( 'should not set needsAutosave=true when editReference changes and the post started autosaving', () => {
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-			wrapper.setProps( {
-				isAutosaving: false,
-				editsReference: [],
-			} );
-			wrapper.setProps( {
-				isAutosaving: true,
-				editsReference: [],
-			} );
-			expect( wrapper.instance().needsAutosave ).toBe( false );
-		} );
+	it( 'should autosave when `editReference` changes', () => {
+		const autosave = jest.fn();
+		const { rerender } = render(
+			<AutosaveMonitor isDirty isAutosaveable autosave={ autosave } />
+		);
+
+		expect( autosave ).toHaveBeenCalledTimes( 0 );
```

Comment:
Would it make more sense to replace these checks:

```
expect( autosave ).toHaveBeenCalledTimes( 0 );
```

with

```
expect( autosave ).not.toHaveBeenCalled()
```

No strong preference, so feel free to leave it as it is.

Reply:
Sure, that's a good call 👍 Done in https://github.com/WordPress/gutenberg/pull/44492/commits/c156af10ea0602d67d009576a012ebc8b1adeafa